src/app.tsx

```diff
@@ -127,47 +128,63 @@ export class AppState extends EventEmitter<AppStateEvents> {
         if (this.#initPromise)
             return this.#initPromise;
 
+        const init_connection = async (connectionName: ConnectionName) => {
+            try {
+                await getLibvirtVersion({ connectionName });
+                const promises = await getApiData({ connectionName });
+                const errorMsgs = promises
+                        .filter(promise => promise.status === 'rejected')
+                        .map(promise => promise.reason.message);
+                if (errorMsgs.length > 0) {
+                    addNotification({
+                        text: _("Failed to fetch some resources"),
+                        detail: errorMsgs.join(', ')
+                    });
+                }
+                // Get the node devices in the background since
+                // they are expensive to get and not important for
+                // displaying VMs.
+                nodeDeviceGetAll({ connectionName }).catch(exc => {
+                    addNotification({
+                        text: "Failed to retrieve node devices",
+                        detail: String(exc),
+                    });
+                });
+            } catch (ex) {
+                // access denied is expected for unprivileged session
+                if (connectionName !== 'system' || superuser.allowed ||
+                    !(ex && typeof ex === 'object' && 'name' in ex && ex.name == 'org.freedesktop.DBus.Error.AccessDenied'))
+                    console.error("Failed to get libvirt version from the dbus API:", ex);
+                /* If the API call failed on system connection and the user has superuser privileges then show the Empty state screen */
+                if (connectionName == "system")
+                    this.systemSocketInactive = true;
+            }
+        };
+
+        const init_hwvirt = async () => {
+            try {
+                const hardwareVirtCheck = await cockpit.script(
+                    "LANG=C.UTF-8 virt-host-validate qemu | grep 'Checking for hardware virtualization'");
+                this.hardwareVirtEnabled = hardwareVirtCheck.includes('PASS');
+            } catch (ex) {
+                // That line doesn't exist on some architectures, so the grep may fail
+                console.debug("Failed to check for hardware virtualization:", ex);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2465-4997d2bc-20251229-174830-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)